quinn-proto/src/shared.rs

```diff
@@ -311,3 +311,31 @@ pub struct ClientOpts {
     pub server_name: String,
     pub crypto: Arc<crypto::ClientConfig>,
 }
+
+/// Stateless reset token
+///
+/// Used for an endpoint to securely communicate that it has lost state for a connection.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct ResetToken([u8; RESET_TOKEN_SIZE]);
+
+impl From<[u8; RESET_TOKEN_SIZE]> for ResetToken {
+    fn from(x: [u8; RESET_TOKEN_SIZE]) -> Self {
+        Self(x)
+    }
+}
+
+impl std::ops::Deref for ResetToken {
```

Comment:
Could we `Deref` in a way that we don't need `[..]` in a few places?

Reply:
I tweaked the `extend` call; can't do much for `==` without adding a significantly larger boilerplatey manual `Eq` impl, I think.